main.go

```diff
@@ -85,11 +85,18 @@ func main() {
 		setupLog.Error(err, "unable to create controller", "controller", "BaseboardManagement")
 		os.Exit(1)
 	}
-	if err = (&controllers.PowerCycleJobReconciler{
+	if err = (&controllers.BMCJobReconciler{
 		Client: mgr.GetClient(),
 		Scheme: mgr.GetScheme(),
 	}).SetupWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create controller", "controller", "PowerCycleJob")
+		setupLog.Error(err, "unable to create controller", "controller", "BMCJob")
+		os.Exit(1)
+	}
+	if err = (&controllers.BMCTaskReconciler{
+		Client: mgr.GetClient(),
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "BMCTask")
 		os.Exit(1)
 	}
```

Comment:
For non-trivial errors, isolate the error check. Its far less cognitive load on maintainers.

```go
err := (&controllers.BMCJobReconciler{
	Client: mgr.GetClient(),
	Scheme: mgr.GetScheme(),
}).SetupWithManager(mgr)
if err != nil {
  // handle
}

Reply:
Actually this was again auto generated by kubebuilder. I hadn't gotten to editing these files yet.
I can definitely change it up 👍

Reply:
agreed, this is auto-generated, so I don't care too much here